manim/debug/scene_debugger.py

```diff
@@ -0,0 +1,221 @@
+import typing
+from functools import wraps
+from inspect import isfunction, ismethod, getmodule
+from types import MappingProxyType
+
+import numpy as np
+from manim.mobject.mobject import Mobject
+from PIL import Image, ImageDraw, ImageFont
+
+from .. import logger
+from ..constants import DOWN, LEFT, RIGHT, UP
+
+__all__ = ["SceneDebugger"]
+
+
+class SceneDebugger:
+    def __init__(self) -> None:
+        self._scene_info = None
+        self._renderer_info = None
+
+        self.debug_animation_attributes = {"run_time", "lag_ratio"}
+
+        self.debug_mobjects_attributes = {"color"}
+
+        self.debug_scene_attributes = {"num_plays", "time"}
+
+        self._record_spied_functions = {}
+        self._force_called_spied_functions = set()
+
+        self.VISUAL_OFFSET = 10
+        self.VECTOR_OFFSET = np.full(2, self.VISUAL_OFFSET)
+
+    def set_scene_vars(self, scene_vars: MappingProxyType) -> None:
+        """Internally used to allow the debugger to access scene debug values. MappingProxy is used to prevent any
+        monkeypatch from happening (sneaky modifications of scene_vars, that could affect the original scene object.)
+
+        Parameters
+        ----------
+        scene_vars : MappingProxyType
+            Variables of the scene.
+        """
+        self._scene_info = scene_vars
+
+    def set_renderer_vars(self, renderer_vars: MappingProxyType) -> None:
+        """Internally used to allow the debugger to access rendered debug values. ``MappingProxy`` is used to prevent any
+        monkeypatch from happening (sneaky modifications of renderer_vars, that could affect the original renderer object.)
+
+        Parameters
+        ----------
+        renderer_vars : MappingProxyType
+            Bariables of the renderer.
+        """
+        self._renderer_info = renderer_vars
+
+    def _get_scene_dict_info(self) -> dict:
+        debug_dict_info = {}
+        # NOTE : There is no way provided by the API to get the current animation.
+        debug_dict_info["Played Animations"] = self._scene_info["animations"]
+        debug_dict_info["time"] = self._renderer_info["time"]
```

Comment:
```suggestion
        debug_dict_info["time"] = round(self._renderer_info["time"], 3)
```
Couldn't we do something like this to make the time look cleaner?

Reply:
Yes; I'll, but the fact it won't be generalized to all of what can be displayed. (there is no general process of automatic wrapping for long variables)